ui/vty-app/Glue.hs

```diff
@@ -82,26 +82,27 @@ knitFaceToUI UiFace{..} KnitFace{..} putPass =
     , langGetHelp = getKnitHelp (Proxy @components)
     }
   where
-    putCommand silent mOp expr = do
+    putCommand ui silent mOp expr = do
       cid <- newUnique
-      fmap (commandIdToUI cid) . putKnitCommand (commandHandle silent mOp cid) $ expr
+      fmap (commandIdToUI cid) . putKnitCommand (commandHandle ui silent mOp cid) $ expr
 
     putUiCommand silent op = case opToExpr op of
       Left err -> return $ Left err
       Right expr -> do
         whenJust (extractPass op) pushPassword
-        comId <- putCommand silent (Just op) expr
-        unless silent $
-          putUiEvent . UiCommandEvent comId . UiCommandWidget $ Knit.ppExpr expr
```

Comment:
How about removing this `UiCommandWidget` as well?

Reply:
How would you do it? We should notify REPL about it somehow. Notifying it directly is complicated, because it can happen in any widget and sending an event from arbitrary widget A to REPL is complicated.
Do you mean replacing it with something else?